### DISCUSSION TITLE
Assistance with Fallback to Local JSON Files

### DISCUSSION THREAD

**Discussion by ahmedghonim** [amannn/next-intl#1276] [Q&A]
Hi 
Hi next-intl Team,

I'm currently working on a project using next-intl for internationalization, and I've run into an issue with setting up a fallback mechanism. Here's what I'm trying to achieve:

1- Primary Goal: Fetch translations from an API based on the user's locale.
2- Fallback Mechanism: If the API doesn't provide translations for a specific locale, I want to fall back to local JSON files stored in the /public/locales directory.

this is my code 
```
export default getRequestConfig(async ({ locale }) => {
  // Validate that the incoming `locale` parameter is valid
  const translations = (await fetchTranslations()) as any;

  if (!translations[locale]) notFound();

  return {
    messages: temperRemoveDupleCurlPresses(translations[locale])
  };
});
```
and this is my local file 
`(await import(`./public/locales/${locale}.json`)).default`

**Comment by amannn** [comment]
What is the specific issue you have? The code above looks like a good starting point, but instead of calling `notFound` when no translations are found, you could read the ones from your JSON file in that case, right?
